odl/util/utility.py

```diff
@@ -163,6 +185,145 @@ def is_complex_floating_dtype(dtype):
     return np.issubsctype(dtype, np.complexfloating)
 
 
+def default_dtype(impl, field):
```

Comment:
Why does it return Rn? It is just a wrapper for Fn with some flags. Would be more simple to just do Fn.